src/main/java/io/github/alathra/alathraskills/api/SkillsPlayerManager.java

```diff
@@ -33,25 +34,25 @@
 import java.util.stream.Stream;
 
 public class SkillsPlayerManager implements Reloadable {
-    private static HashMap<UUID, SkillsPlayer> skillPlayers = new HashMap<UUID, SkillsPlayer>();
+    private static HashMap<UUID, SkillsPlayer> skillPlayers = new HashMap<>();
     private final AlathraSkills instance;
     private BukkitTask storePlayerSkillInfoTask;
 
     public SkillsPlayerManager(AlathraSkills instance) {
         this.instance = instance;
-        SkillsPlayerManager.skillPlayers = new HashMap<UUID, SkillsPlayer>();
+        SkillsPlayerManager.skillPlayers = new HashMap<>();
     }
 
-    // TODO Note, try to cut down on the amount of queries executed per player
-    public static CompletableFuture<SkillsPlayer> handlePlayerJoin(Player p) {
+    public CompletableFuture<SkillsPlayer> handlePlayerJoin(Player p) {
         return CompletableFuture.supplyAsync(() -> {
             HashMap<Integer, SkillDetails> playerSkills = new HashMap<>();
             HashMap<Integer, Float> playerExperienceValues = new HashMap<>();
 
-            if (!p.hasPlayedBefore()) {
-                DatabaseQueries.saveAllSkillCategoryExperience(p, 0.f, 0.f, 0.f);
-                DatabaseQueries.savePlayerData(p, 0, 0, Instant.now());
-            }
+            // TODO: test if this needs to stay or not.
```

Comment:
Is this needed?

Reply:
Right, this was the thing I didn't get araound to testing - however looking at the whole method it shouldn't be necessary to first save to DB before loading, as the values are assigned to 0 before they are fetched. I'll remove the commented code as well.